src/main/scala/fm/xml/XmlReader.scala

```diff
@@ -17,20 +17,57 @@ package fm.xml
 
 import com.ctc.wstx.stax.WstxInputFactory
 import fm.common.Implicits._
-import fm.common.{Logging, Resource}
+import fm.common.{InputStreamResource, Logging, Resource, SingleUseResource}
 import fm.lazyseq.ResourceLazySeq
-import java.io.{InputStream, Reader, StringReader}
+import fm.xml.RichXMLStreamReader2.toRichXMLStreamReader2
+import java.io.{File, InputStream, Reader, StringReader}
 import javax.xml.bind.{JAXBContext, Unmarshaller}
 import javax.xml.stream.XMLInputFactory
 import javax.xml.stream.XMLStreamConstants.START_ELEMENT
 import org.codehaus.stax2.XMLStreamReader2
 import org.codehaus.stax2.util.StreamReader2Delegate
-import scala.reflect.{ClassTag, classTag}
-import RichXMLStreamReader2.toRichXMLStreamReader2
+import scala.reflect.{classTag, ClassTag}
 import scala.util.Try
 
 object XmlReader {
 
+  case class MultiXmlReaderConstructor[T: ClassTag](
```

Comment:
What's going on here?

Reply:
Why is `: ClassTag` being used here?

Reply:
These helpers probably aren't needed, I think I had them in there for some testing purposes before I simply integrated it into the `XmlReaderWriter`

Reply:
I moved/simplified these into a `MultiXmlReaderWriter`